### ISSUE TITLE
[Relay][Bug] nn.batch_norm not being simplified by SimplifyInference

### ISSUE THREAD

**Issue by gussmith23** [apache/incubator-tvm#6852]
cc @hypercubestart, who discovered this oddity

```python
r = """
#[version = "0.0.5"]
def @main(%data: Tensor[(1, 32, 32, 16), float32], %bn_gamma: Tensor[(16), float32], %bn_beta: Tensor[(16), float32], %bn_mean: Tensor[(16), float32], %bn_var: Tensor[(16), float32]) -> (Tensor[(1, 32, 32, 16), float32], Tensor[(16), float32], Tensor[(16), float32]) {
    nn.batch_norm(%data, %bn_gamma, %bn_beta, %bn_mean, %bn_var, axis=3) /* ty=(Tensor[(1, 32, 32, 16), float32], Tensor[(16), float32], Tensor[(16), float32]) */
}
"""
mod = tvm.parser.parse(r)
mod = relay.transform.SimplifyInference()(mod)
print(mod.astext())
```
the result is not what we expected, though, as the batch norm doesn’t get simplified:
```
#[version = "0.0.5"]
def @main(%data: Tensor[(1, 32, 32, 16), float32], %bn_gamma: Tensor[(16), float32], %bn_beta: Tensor[(16), float32], %bn_mean: Tensor[(16), float32], %bn_var: Tensor[(16), float32]) -> (Tensor[(1, 32, 32, 16), float32], Tensor[(16), float32], Tensor[(16), float32]) {
  nn.batch_norm(%data, %bn_gamma, %bn_beta, %bn_mean, %bn_var, axis=3) /* ty=(Tensor[(1, 32, 32, 16), float32], Tensor[(16), float32], Tensor[(16), float32]) */
}
```
is that to be expected? are we doing something dumb here?

**Comment by gussmith23** [comment]
Nevermind, I think this is to be expected. It's because we weren't indexing into the tuple result/we hadn't changed the return type of the function.

```python
import tvm
from tvm import relay

r = """
#[version = "0.0.5"]
def @main(%data: Tensor[(1, 32, 32, 16), float32], %bn_gamma: Tensor[(16), float32], %bn_beta: Tensor[(16), float32], %bn_mean: Tensor[(16), float32], %bn_var: Tensor[(16), float32]) -> Tensor[(1, 32, 32, 16), float32] {
    %0 = nn.batch_norm(%data, %bn_gamma, %bn_beta, %bn_mean, %bn_var, axis=3) /* ty=(Tensor[(1, 32, 32, 16), float32], Tensor[(16), float32], Tensor[(16), float32]) */;
%0.0
}
"""
mod = tvm.parser.parse(r)
mod = relay.transform.SimplifyInference()(mod)
print(mod.astext())
```

**Comment by comaniac** [comment]
The root cause is that the simplify pass detects the `batch_norm -> get()` pattern, so it would ignore the batch_norm without get().

**Comment by gussmith23** [comment]
Makes sense. We hit a weird corner case here, glad we figured it out. Maybe a documentation change is warranted, but it's also a pretty niche corner case, so I'm not sure it matters!

**Comment by comaniac** [comment]
It doesn't hurt to improve the document for this case as we met this issue as well several months ago lol
Please let me know if you would file a PR; otherwise I will put it in my TODO list but I'm not sure when I will do so..

**Comment by gussmith23** [comment]
See the above! Let me know if the wording is okay...
